app/battle_preparation/knight.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from app.battle_preparation.armour import Armour
+from app.battle_preparation.potion import Potion
+from app.battle_preparation.weapon import Weapon
+
+
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = Armour(armour)
+        self.weapon = Weapon(weapon)
+        self.potion = Potion(potion)
+        self.protection = self.armour.protection
+
+    @classmethod
+    def dict_read(cls, dict_knights: dict) -> Knight:
```

Comment:
Why did you create this method? You can use __init__ without this method.

Reply:
I created this method to have an alternative method to create knights directly from the dictionary. Yes, I can do the same thing in the init method, but then I won't be able to create a single knight at a time

Reply:
Why?

Reply:
I don't know I only study this stuff.